extensions/amp-live-list/0.1/amp-live-list.js

```diff
@@ -161,6 +169,15 @@ export class AmpLiveList extends AMP.BaseElement {
     /** @private {?Element} */
     this.pendingPagination_ = null;
 
+    /**
+     * Element that dynamically built this amp-live-list (if any).
+     * @private {?Element}
+     */
+    this.listBuilder_ = null;
+
+    /** @private {string} */
+    this.listBuilderId_ = '';
```

Comment:
If I'm understanding this PR correctly, then I think this name is a bit confusing.  This is the ID referred to in the `built-by` attribute, right?  The name would suggest it's the ID of the `this.listBuilder_` element.

Reply:
Yes, those two are the same actually. Whenever the "listBuilder" creates the `live-list`, they will specify their own `id` in the `built-by` attribute of the `live-list`. Does that make sense?

Reply:
Well, kind of.  I guess I see two scenarios:

1. **We trust the "list builder" to set the correct ID.** . If we trust this, then there's not really any need to actually store the `this.listBuilder_` element, right?  All we ever do is check for its existence, which would be synonymous with checking for the presence of the ID, since we're trusting that the ID is correct.
2. **We *don't* trust the "list builder" to set the correct ID.** If we don't trust this, then these two things are not the same.  The ID is what the list builder claims to be correct, but we have not actually verified is correct, which would be good to call out either in the name, or the docs, or both.  In other words, if the list builder sets the wrong ID, then it's confusing to say "the ID of the list builder, and the list builder ID are different."

WDYT?

Reply:
That's true. In theory, the "list builder" could specify any ID, regardless if it was the element that built the list or not, or if it even exists.

I'll change the names and hope it will make it clearer.